processing/src/main/java/io/druid/query/QueryToolChest.java

```diff
@@ -46,16 +45,16 @@
   public abstract QueryRunner<ResultType> mergeResults(QueryRunner<ResultType> runner);
 
   /**
-   * Creates a builder that is used to generate a metric for this specific query type.  This exists
-   * to allow for query-specific dimensions on metrics.  That is, the ToolChest is expected to set some
+   * Creates a {@link QueryMetrics} object that is used to generate metrics for this specific query type.  This exists
+   * to allow for query-specific dimensions and metrics.  That is, the ToolChest is expected to set some
    * meaningful dimensions for metrics given this query type.  Examples might be the topN threshold for
    * a TopN query or the number of dimensions included for a groupBy query.
    *
    * @param query The query that is being processed
    *
-   * @return A MetricEvent.Builder that can be used to make metrics for the provided query
+   * @return A QueryMetrics that can be used to make metrics for the provided query
    */
-  public abstract ServiceMetricEvent.Builder makeMetricBuilder(QueryType query);
+  public abstract QueryMetrics<? super QueryType> makeMetrics(QueryType query);
```

Comment:
Is it expected that any extensions adding query types should use an injected `QueryMetricsFactory` to create this instance, rather than do it from scratch?